src/items/static-items.md

```diff
@@ -8,16 +8,17 @@
 > [^extern-safety]: The `safe` and `unsafe` function qualifiers are only
 >   allowed semantically within `extern` blocks.
 
-A *static item* is similar to a [constant], except that it represents a precise
-memory location in the program. All references to the static refer to the same
-memory location. Static items have the `static` lifetime, which outlives all
-other lifetimes in a Rust program. Static items do not call [`drop`] at the
-end of the program.
+A *static item* is similar to a [constant], except that it represents an allocated object in the
+program that is initialized with the initializer expression. This allocated object is disjoint from
+all other allocated objects. All references and raw pointers to the static refer to the same
```

Comment:
Do we want to guarantee that they are disjoint from other allocated objects? Or just other statics. IE. if I have:
```rust
static FOO: i32 = 0;
const BAR: i32 = 0;

fn foo(){
    assert!(!core::ptr::eq(&FOO, &BAR));
}
```

Should we guarantee the assertion will always pass?

Reply:
I don't have concern permissions, but I think this should be addresses by T-opsem before the end of the FCP.

Reply:
Yeah I think we should guarantee that, and it's what the text already says, isn't it?

Reply:
That prevents emitting `unnamed_addr` on `const` items.

Reply:
`unnamed_addr` specifically *does* coalescing of non-significant-address-items into each other, and potentially into a significant-address item.

Reply:
Wait, LLVM will merge `unnamed_addr` allocations with "named addr" (non-`unnamed_addr`) allocations? That's... surprising.

Cc @nikic

Reply:
The LangRef says:
> Note that a constant with significant address can be merged with a unnamed_addr constant, the result being a constant whose address is significant.

Reply:
Ah, that is unfortunate.

Reply:
> Yeah I think we should guarantee that, and it's what the text already says, isn't it?

That's what the proposed text says, my question is whether that's what we want it to say.

Reply:
I... honestly don't think that's surprising at all?

That's exactly what I would expect in a case of
```rust
const BIG_CONST: BigFrozen = BigFrozen::big_init();
static BIG_STATIC: BigFrozen = BIG_CONST;
```

That's the precise situation where the `const` will be unified "into" the `static`.

Reply:
> I... honestly don't think that's surprising at all?

I guess we have different intuitions then.

But reality clearly works one way here, so I have updated the text to match reality. @chorman0773 please have a look.